src/common/snippets/src/lowered/shape_inference/shape_inference.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "snippets/lowered/shape_inference/shape_inference.hpp"
+#include "snippets/lowered/shape_inference/shape_infer_instances.hpp"
+#include <openvino/op/util/unary_elementwise_arithmetic.hpp>
+#include <openvino/op/util/binary_elementwise_arithmetic.hpp>
+#include <openvino/op/util/binary_elementwise_comparison.hpp>
+#include <openvino/op/util/binary_elementwise_logical.hpp>
+#include <openvino/op/parameter.hpp>
+#include <openvino/op/result.hpp>
+#include <snippets/snippets_isa.hpp>
+
+namespace ov {
+namespace snippets {
+using ShapeInferPtr = IShapeInferSnippetsFactory::ShapeInferPtr;
+
+ShapeInferPtr IShapeInferSnippetsFactory::make(const ov::DiscreteTypeInfo& key, const std::shared_ptr<ov::Node>& op) {
+    const auto& maker_iter = registry.find(key);
+    if (maker_iter != registry.end())
+        return maker_iter->second(op);
+    return get_specific_op_shape_infer(key, op);
+}
+
+ShapeInferPtr IShapeInferSnippetsFactory::get_specific_op_shape_infer(const ov::DiscreteTypeInfo& key,
+                                                                      const std::shared_ptr<ov::Node>& op) const {
+    return {};
+}
+
+#define SHAPE_INFER_PREDEFINED(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>();} }
+#define SHAPE_INFER_OP_SPECIFIC(OP) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<OP::ShapeInfer>(n);} }
+#define SHAPE_INFER_OP_SPECIFIC_EXTERNAL(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>(n);} }
+
+const IShapeInferSnippetsFactory::TRegistry IShapeInferSnippetsFactory::registry {
+        SHAPE_INFER_PREDEFINED(op::ConvertTruncation, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(op::ConvertSaturation, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(op::Load, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(op::Store, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(op::Buffer, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(op::Fill, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(ov::op::v0::Parameter, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(ov::op::v1::Softmax, PassThroughShapeInfer),
+        SHAPE_INFER_PREDEFINED(ov::op::v8::Softmax, PassThroughShapeInfer),
```

Comment:
BTW could you please add comment that it should be removed when we move decomposition to data flow in bounds of task 112847? I think we will forget it 😃  Thank you!

Reply:
No problem, done